more_itertools/more.py

```diff
@@ -3597,9 +3597,10 @@ def _sample_weighted(iterator, k, weights, strict):
         else:
             weights_to_skip -= weight
 
-    # Equivalent to [element for weight_key, element in sorted(reservoir)]
-    shuffle(reservoir)
-    return [heappop(reservoir)[1] for _ in range(k)]
+    # Equivalent to [element for weight_key, element i n sorted(reservoir)]
+    ret = [heappop(reservoir)[1] for _ in range(k)]
```

Comment:
Since this gets shuffled later, there is no need for ordering with *heappop()*. 

This line can be simplified to:

`ret = [element for weight_key, element in reservoir]`

Reply:
Hmm, isn't the point of doing the `heappop` to get the first `k` as ordered by the preceeding steps?

Reply:
Double check me on this, but it looks like reservoir is always of size *k*.  The `take()` step puts in *k* elements. The `heapify` step does not change the size, nor does the `heapreplace()` step.

So the "first k" in the reservoir is the same as the entire reservoir.

Reply:
Ah, you're right! Thanks - I stand corrected.